compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -6689,6 +6695,12 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           method match {
             case js.StringLiteral(value) =>
               if (js.JSGlobalRef.isValidJSGlobalRefName(value)) {
+                if (value == "await") {
+                  reporter.warning(pos,
+                      s"Calling a method of the global scope with the name '$value' is deprecated.\n" +
+                      "  It may produce invalid JavaScript code causing a SyntaxError in some environments." +
+                      GenericGlobalObjectInformationMsg)
+                }
```

Comment:
Somewhat unrelated: But it seems like there is potential for refactoring here for all the error cases (`resolveGlobalScope` or something).

Reply:
I managed to factor out the logic.